### PR TITLE
Adding roles sample

### PR DESCRIPTION
**Pull request by derisen**
# Pull Request

## Purpose

* This PR adds a sample demonstrating how to use Azure AD App roles in a confidential client web app using MSAL Node

## Does this introduce a breaking change

```console
    [ ] Yes
    [ x ] No
```

## Pull request type

What kind of change does this Pull Request introduce?

```console
    [ ] Bugfix
    [ x ] Feature
    [ ] Code style update (formatting, local variables)
    [ ] Documentation content changes
    [ ] Other... Please describe:
```

## How to test

* Get the code

```console
    git clone https://github.com/Azure-Samples/ms-identity-javascript-nodejs-tutorial.git
    cd ms-identity-javascript-nodejs-tutorial
    git checkout derisen
    cd 4-AccessControl/1-app-roles/App
    npm install
```

## What to check

ex: verify that the following are valid:

* Run in Chrome, Edge or Firefox
* Run in private mod

### CODE REVIEW

File: 4-AccessControl/1-app-roles/README.md [hunk 1]
```diff
@@ -0,0 +1,326 @@
+# A Node.js & Express web app calling Microsoft Graph using App Roles to implement Role-Based Access Control
```
Comment (kalyankrishna1) [inline 4-AccessControl/1-app-roles/README.md:1]: >calling Microsoft Graph [](http://example.com/codeflow?start=30&length=23)

"calling Microsoft Graph" can be dropped

File: 4-AccessControl/1-app-roles/README.md [hunk 2]
```diff
@@ -0,0 +1,326 @@
...
+ 1. [About the code](#about-the-code)
+ 1. [More information](#more-information)
+ 1. [Community Help and Support](#community-help-and-support)
+ 1. [Contributing](#contributing)
+
+![Build badge](https://identitydivision.visualstudio.com/_apis/public/build/definitions/a7934fdd-dcde-4492-a406-7fad6ac00e17/<BuildNumber>/badge)
+
+## Overview
+
+This sample demonstrates a Node.js & Express web app featuring a todo list and secured with the [Microsoft Authentication Library for Node.js](). The app implements **Role-based Access Control** (RBAC) by using Azure AD **App Roles**. In the sample, a **dashboard** component allows signed-in users to see the tasks assigned to them and is only accessible by users under an **app role** named **TaskAdmin**.
```
Comment (kalyankrishna1) [inline 4-AccessControl/1-app-roles/README.md:20]: >](). [](http://example.com/codeflow?start=141&length=4)

link missing

Comment (derisen) [reply 4-AccessControl/1-app-roles/README.md]: done

**Review by kalyankrishna1 — APPROVED** [review]
:shipit:

File: 4-AccessControl/2-security-groups/README.md [hunk 3]
```diff
@@ -0,0 +1,332 @@
...
+| `AppCreationScripts/`       | Contains Powershell scripts to automate app registration.     |
+| `ReadmeFiles/`              | Contains illustrations and screenshots.                       |
+| `App/appSettings.json`      | Authentication parameters and settings.                       |
+| `App/data/cache.json`       | Stores MSAL Node token cache data.                            |
+| `App/data/db.json`          | Stores todo list data.                                        |
+| `App/app.js`                | Application entry point.                                      |
+
+## Prerequisites
+
+- [Node.js](https://nodejs.org/en/download/) must be installed to run this sample.
```
Comment (idg-sam) [inline 4-AccessControl/2-security-groups/README.md:44]:
```suggestion
- [Node.js](https://nodejs.org/en/download/) version => 12 must be installed to run this sample.
```

**Review by idg-sam — APPROVED** [review]
Looks good. Reame is good. Maybe lets' also add groups claim to the list of displayed claims? Also consider a link from dashboard to todolist, and a link from todolist to dashboard

**Pull request merged**